client-docs/datatype.md

```diff
@@ -0,0 +1,226 @@
+
+const mapeo = createClient({
+  dataTypes: {
+    observation: ObservationSchema,
+    preset: PresetSchema,
+  },
+});
+
+// Now you can use the CRUD methods with the
+// mapeo.observation and mapeo.preset namespaces!
+```
+
+## Types
+
+### `MapeoDoc`
+
+Basically represents the entity that’s stored in the database. Has information about the specific application data in its `value` field, as well as some meta information about the [`Doc`](https://github.com/digidem/mapeo-core-next/blob/main/lib/types.js#L90) as top level fields, such as `id`, `version`, `deleted`, `links`, `forks`, `created_at`, and `updated_at`.
+
+### `MapeoType`
+
+Represents any application-specific key-value data associated with a `MapeoDoc`. This can be updated in the database and is generally available on a doc as `doc.value`
+
+## Methods
+
+### `create`
+
+`(value: MapeoType) => Promise<MapeoDoc<MapeoType>>`
+
+Create a document with the associated `value`. Resolves with the Mapeo doc containing information created upon saving to the database.
+
+```ts
+// Let's say our observation data has this shape
+type Observation = {
+  lat: number;
+  lon: number;
+  tags: { type: string };
+};
+
+// We can create an observation document like this
+const doc = await mapeo.observation.create({
+  lat: 0,
+  lon: 0,
+  tags: { type: "animal" },
+});
+
+// We can access the data we specified using `doc.value`
+console.log(doc.value); // prints { lat: 0, lon: 0, tags: { type: "animal" } }
+```
+
+### `getByDocId`
+
+`(docId: string, opts?: { deleted?: boolean }) => Promise<MapeoDoc<MapeoType>>`
+
+Get a document with the associated `docId`. `opts.deleted` can be specified to include or exclude a result based on whether the document was deleted or not. By default, `opts.deleted` is `true`.
+
+```ts
+// Create a doc
+const createdDoc = await mapeo.observation.create({ ... })
+
+// Get the doc using the id we have after creating it
+const retrievedDoc = await mapeo.observation.getByDocId(createdDoc.id)
+
+// Can guarantee that this passes
+console.assert(createdDoc.id === retrievedDoc.id)
+```
+
+### `getByVersionId`
+
+`(versionId: string, opts?: { deleted?: boolean }) => Promise<MapeoDoc<MapeoType>>`
+
+Get a document with the associated `versionId`. `opts.deleted` can be specified to include or exclude a result based on whether the document was deleted or not. By default, `opts.deleted` is `true`.
+
+```ts
+// Create a doc
+const createdDoc = await mapeo.observation.create({ ... })
+
+// Get the doc using the id we have after creating it
+const retrievedDoc = await mapeo.observation.getByVersionId(createdDoc.version)
+
+// Can guarantee that this passes
+console.assert(createdDoc.version === retrievedDoc.version)
+```
+
+### `getMany`
+
+`(opts?: { deleted?: boolean, limit?: number}) => Promise<MapeoDoc<MapeoType>[]>`
+
+Get many documents, sorted by `created_at` descending. If no matching documents exist, resolves with an empty array. Can accept the following options (`opts`):
```

Comment:
```suggestion
Get many documents, sorted by `created_at` descending (most recent documents first). If no matching documents exist, resolves with an empty array. Can accept the following options (`opts`):
```
It always takes me a moment to parse what "descending" means.

Reply:
addressed in aab35fcf3c6eca5feccaefb8ebf669982e9dc31d